sprocs/questions_select.sql

```diff
@@ -1,15 +1,44 @@
-CREATE FUNCTION
+DROP FUNCTION IF EXISTS questions_select(bigint);
+DROP FUNCTION IF EXISTS questions_select(bigint, bigint);
+
+CREATE OR REPLACE FUNCTION
     questions_select (
         IN question_id bigint,
-        OUT question questions
+        IN instance_question_id bigint,
+        OUT question jsonb
     )
 AS $$
 BEGIN
-    SELECT *
-    INTO question
-    FROM questions
-    WHERE id = question_id;
+    IF instance_question_id IS NULL THEN
+        SELECT to_jsonb(q.*) 
+            || jsonb_build_object('question_params', pc.question_params) 
+        INTO question
+        FROM questions AS q
+        JOIN pl_courses AS pc ON q.course_id = pc.id
+        WHERE q.id = question_id;
+
+        IF NOT FOUND THEN RAISE EXCEPTION 'no such question_id: %', question_id; END IF;
+    ELSE 
+        SELECT to_jsonb(q.*) 
+            || jsonb_build_object('question_params', pc.question_params 
+                || ci.question_params
+                || aset.question_params 
+                || z.question_params 
+                || ag.question_params 
+                || aq.question_params)
```

Comment:
If `question_params` are NULL you can use `COALESCE(??.question_params, '{}'::jsonb)` instead.

Reply:
I *really* don't like that `{some jsonb} || NULL` gives `NULL`. Alas, this is fixed.